### PR TITLE
Cache trusted cert values, invalidating when anything changes

### PR DESCRIPTION
**Pull request by sgmiller**
Previously, trusted certs were loaded and parsed on every login call.  This
is both slow, and means a copy of the trusted certs is in memory per login. 
For highly concurrent use this raises Vault's memory usage unnecessarily.

### CODE REVIEW

**Comment by sgmiller** [comment]
Convering to draft to add cache sizing and unit tests, docs

File: builtin/credential/cert/path_login.go [hunk 1]
```diff
@@ -646,6 +657,13 @@ func (b *backend) loadTrustedCerts(ctx context.Context, storage logical.Storage,
 			conf.QueryAllServers = conf.QueryAllServers || entry.OcspQueryAllServers
 		}
 	}
+
+	b.trustedCache.Add(certName, &trusted{
```
Comment (stevendpclark) [inline builtin/credential/cert/path_login.go:661]: nit: Should this only happen `if !b.trustedCacheDisabled`

**Review by stevendpclark — APPROVED** [review]
👍

File: builtin/credential/cert/path_login.go (hunk 1 above)
Comment (sgmiller) [reply builtin/credential/cert/path_login.go:661]: Definitely, otherwise we're just wasting memory.

**Review by stevendpclark — APPROVED** [review]
Good catch on the atomic bool, completely missed that

**Pull request merged**